crates/sui-core/src/node_sync/node_state.rs

```diff
@@ -584,9 +585,7 @@ where
         };
         match result {
             Ok(_) => Ok(SyncStatus::CertExecuted),
-            Err(SuiError::ObjectNotFound { .. })
-            | Err(SuiError::ObjectErrors { .. })
-            | Err(SuiError::SharedObjectLockNotSetError) => {
+            Err(SuiError::ObjectNotFound { .. }) | Err(SuiError::ObjectErrors { .. }) => {
```

Comment:
Note that the previous inclusion of SharedObjectLockNotSetError was an error - it is previously only ever used as a nested error for the ObjectErrors type. Now we treat it as a "cannot continue" signal.

Reply:
Which logic eventually receives the `SuiError::SharedObjectLockNotSetError`, and if / how it will be retried?

Reply:
And just to make sure, now with SuiError::SharedObjectLockNotSetError we are avoiding the recursive parent execution, and presumably the cert will be retried with some delay?

Reply:
Chatted with Mark off thread. The retry is [here](https://github.com/MystenLabs/sui/blob/d8633441e79f3d30aba4b001de94808c294b4317/crates/sui-core/src/authority_active/execution_driver/mod.rs#L71). Most likely the `SharedObjectLockNotSetError` will add a 500ms delay between retries.

Reply:
that's correct - but crucially this prunes entire subtrees of shared objects at once, because an unsequenced cert cannot have sequenced children.